tests/utils/test_file.py

```diff
@@ -212,3 +213,26 @@ def test_get_modules_from_invalid_file(self):
         modules = list(file_utils.iter_airflow_imports(file_path))
 
         assert len(modules) == 0
+
+
+def test_get_unique_dag_module_name():
+    edge_filenames = [
+        "test_dag.py",
+        "test-dag.py",
+        "test-dag-1.py",
+        "test-dag_1.py",
+        "test-dag.dev.py",
+        "test_dag.prod.py",
+    ]
+    # sha1 of file_path in middle
+    expected_regex = [
+        r"unusual_prefix_[0-9a-f]{40}_test_dag",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_1",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_1",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_dev",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_prod",
+    ]
+    for idx, filename in enumerate(edge_filenames):
+        modify_module_name = file_utils.get_unique_dag_module_name(filename)
+        assert re2.match(re2.compile(expected_regex[idx]), modify_module_name) is not None
```

Comment:
How about mocking the hexdigest of the `hashlib.sha1` output like this?

```suggestion
@pytest.mark.parametrize(
    "edge_filenames, expected_module_name",
    [
        ("test_dag.py", "unusual_prefix_mocked_sha1_test_dag"),
        ("test-dag.py", "unusual_prefix_mocked_sha1_test_dag"),
        ("test-dag-1.py", "unusual_prefix_mocked_sha1_test_dag_1"),
        ("test-dag_1.py", "unusual_prefix_mocked_sha1_test_dag_1"),
        ("test-dag.dev.py", "unusual_prefix_mocked_sha1_test_dag_dev"),
        ("test_dag.prod.py", "unusual_prefix_mocked_sha1_test_dag_prod"),
    ],
)
def test_get_unique_dag_module_name(edge_filenames, expected_module_name):
    with mock.patch("hashlib.sha1") as mock_hash_object:
        mock_hash_object.hexdigest.return_value = "mocked_sha1"
        for filename in enumerate(edge_filenames):
            modify_module_name = file_utils.get_unique_dag_module_name(filename)
            assert expected_module_name == modify_module_name
```

Reply:
Cool, I never thought of this mock approach. I will take this solution. Thanks!

Reply:
I guess you can click `Add suggestion to batch` directly to reduce some efforts lol 
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

Reply:
resolve: laverage `mock.patch` to mock `hashlib.sha1` and directly compare with expected output